src/commands/merge.ts

```diff
@@ -0,0 +1,55 @@
+import { commands, window } from "vscode";
+import { IBranchItem } from "../common/types";
+import { selectBranch } from "../helpers/branch";
+import { Repository } from "../repository";
+import { Command } from "./command";
+
+export class Merge extends Command {
+  constructor() {
+    super("svn.merge", { repository: true });
+  }
+
+  public async execute(repository: Repository) {
+    const branch = await selectBranch(repository);
+
+    if (!branch) {
+      return;
+    }
+
+    await this.merge(repository, branch);
+  }
+
+  async merge(repository: Repository, branch: IBranchItem) {
+    let reintegrate = false;
+    if (repository.currentBranch == "trunk") {
```

Comment:
This will not work for people that don't call their main branch trunk

Reply:
https://github.com/JohnstonCode/svn-scm/blob/d7b26e69b5260d1dccbe6f1e241879a20b513e7e/src/helpers/branch.ts#L9

Reply:
This might help

Reply:
Good point. Added a isTrunk function to branches, that uses the same logic as in getBranchName.